poe_client/rate_limiter.py

```diff
@@ -106,27 +112,27 @@ async def parse_headers(self, headers) -> None:
                 for rule in headers["X-Rate-Limit-{0}".format(rule_name)].split(","):
                     hits, period, restriction = rule.split(":")
 
-                    if policy_id not in self.policies:
-                        self.policies[policy_id] = {}
-
-                    self.policies[policy_id][period] = Policy(
+                    self.policies.setdefault(policy_id, {})[period] = Policy(
                         rule,
                         int(hits),
                         int(period),
                         int(restriction),
                     )
 
                 for state in headers["X-Rate-Limit-{0}-State".format(rule_name)].split(
-                    ","
+                    ",",
                 ):
-                    await self.policies[policy_id][state.split(":")[1]].update_state(
-                        int(state.split(":")[0]),
-                        int(state.split(":")[2]),
+                    hits, period, restriction = state.split(":")
+                    await self.policies[policy_id][period].update_state(
+                        int(hits),
+                        int(restriction),
                     )
 
+        return policy_name
+
     async def get_semaphore(self, policy_name: str) -> bool:
         """Get a semaphore to make a request."""
-        if len(self.policies) == 0:
+        if not len(self.policies):
```

Comment:
could just be `if not` right? An empty list should be falsey.

Reply:
Yup, done.